### PR TITLE
Add update instance method and spec

### PR DESCRIPTION
**Pull request by mlan080**
Rspec error: Expected 1 and got 'pandy'

Question: 
1. In IRB, I have managed to update db row id 15 to 'pandy', however output is 1, which means one record has been changed. How do I get the output to return a Shop object?

### CODE REVIEW

File: app/models/shop.rb [hunk 1]
```diff
@@ -24,4 +29,13 @@ def self.last
   def self.all
     DB[:shops].all
   end
+
+  def self.find(id)
+    h = DB[:shops].first(:id => id) #sequel doc uses first for retrieving record
+    Shop.new(h) #returns a shop object with hash argument
+  end
+
+  def set(id, name)
```
Comment (lucasrenan) [inline app/models/shop.rb]: maybe we can call it `update` and no parameter is needed here.

if you create a method reader for `id`, then we can expect a behavior like this:

```ruby
s = Shop.find(1) # let's suppose it has name: "Tea Shop"
s.name = "Updated Shop"
s.update
```

File: app/models/shop.rb [hunk 2]
```diff
@@ -14,20 +14,26 @@ def initialize(hash = {})
...
+    DB[:shops].order(:id).last
   end
 
   def self.all
     DB[:shops].all
   end
 
   def self.find(id)
-    record = DB[:shops].first(id: id) #sequel doc uses first for retrieving record
+    h = DB[:shops].first(:id => id) #sequel doc uses first for retrieving record
```
Comment (lucasrenan) [inline app/models/shop.rb]: record

File: app/models/shop.rb [hunk 3]
```diff
@@ -3,9 +3,9 @@
 DB = Sequel.connect('sqlite://tea-shops.sqlite')
 
 class Shop
-  attr_accessor :name, :description #instance methods
+  attr_accessor :name, :description
```
Comment (lucasrenan) [inline app/models/shop.rb]: `attr_reader :id`

File: app/models/shop.rb [hunk 4]
```diff
@@ -3,9 +3,9 @@
...
   def create
-    shops = DB[:shops] # Define a dataset, table: shops
+    shops = DB[:shops]  
     shops.insert(:name => name, :description => description) # Populate the table, :name is column name, name is method accessing the instance variable
   end
```
Comment (lucasrenan) [inline app/models/shop.rb]: then you can add the id on `initialize`, like `@id = hash[:id]`

File: spec/shop_spec.rb [hunk 5]
```diff
@@ -35,30 +33,32 @@
...
     it 'returns the instance class' do
       result = Shop.find(last_shop[:id])
 
       expect(result.class).to eq(Shop)
     end
+
+  describe "#update" do
+    let(:shop_id) { Shop.new({name: 'Mandy', description: 'be careful'}).create }
+
+    it'should return updated name Pandy in the database' do
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: maybe a better spec description would be `it 'updates the shop'`.

File: spec/shop_spec.rb [hunk 6]
```diff
@@ -35,30 +33,32 @@
...
-      result = Shop.find(15)
-      
-      expect(result.name).to eq(shop.name)
-    end
-  end
-
-  describe "#update" do
     before { Shop.new({name: 'cafe', description: 'red apples'}).create }
 
     let(:last_shop) { Shop.all.last } #defining variable to use in spec
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: you can get the opportunity and replace this part with the same as update `let(:shop_id) { Shop.new({name: 'Mandy', description: 'be careful'}).create }` as ` create` return the shop id created.

Comment (mlan080) [reply spec/shop_spec.rb]: Do you mean get rid of last_shop completely with something like this? 

    describe ".find" do
      let(:shop_id) { Shop.new({name: 'Mandy', description: 'be careful'}).create }

      it 'returns the given shop row from shops table' do
        shop = Shop.find(last_shop[:id])
        shop.name = 'Dandy'
        expect(shop.name).to eq('Dandy')

File: spec/shop_spec.rb [hunk 7]
```diff
@@ -28,37 +28,39 @@
 
     it 'returns all rows from the shops table' do
 
-      expect(Shop.all).to eq(rows)
+    expect(Shop.all).to eq(rows)
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: indentation here

File: spec/shop_spec.rb [hunk 8]
```diff
@@ -28,37 +28,39 @@
...
   end
 
   describe ".find" do
-    before { Shop.new({name: 'cafe', description: 'red apples'}).create }
-
-    let(:last_shop) { Shop.all.last } #defining variable to use in spec
+    let(:shop_id) { Shop.new({name: 'Mandy', description: 'be careful'}).create }
 
     it 'returns the given shop row from shops table' do
+      last_shop = Shop.all.last
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: I think you misunderstood here, I meant something like:

```ruby
let(:shop_id) { Shop.new({name: 'Name', description: 'be careful'}).create }

it 'returns the given shop row from shops table' do
  result = Shop.find(shop_id)

  expect(result.name).to eq('Name')
end

it 'returns a Shop instance' do
  result = Shop.find(shop_id)

  expect(result.class).to eq(Shop)
end
```

File: configs/environments/development.rb [hunk 9]
```diff
@@ -0,0 +1 @@
+development.rb
```
Comment (lucasrenan) [inline configs/environments/development.rb]: this file is not related to this PR

File: configs/environments/production.rb [hunk 10]
```diff
@@ -0,0 +1 @@
+production.rb
```
Comment (lucasrenan) [inline configs/environments/production.rb]: this file is not related to this PR

File: configs/environments/test.rb [hunk 11]
```diff
@@ -0,0 +1 @@
+test.rb
```
Comment (lucasrenan) [inline configs/environments/test.rb]: this file is not related to this PR
